### PR TITLE
Id minter switch to mysql

### PR DESCRIPTION
**Pull request by alicefuzier**
### What is this PR trying to achieve?
Make the id minter not create two identifiers for the same miroID. Related to #343.
### Who is this change for?
Everyone
### Have the following been considered/are they needed?

- [ ] Deployed new versions

### CODE REVIEW

File: common/src/main/scala/uk/ac/wellcome/models/Work.scala [hunk 1]
```diff
@@ -3,18 +3,28 @@ package uk.ac.wellcome.models
 import com.fasterxml.jackson.annotation.JsonProperty
 import com.sksamuel.elastic4s.Indexable
 import uk.ac.wellcome.utils.JsonUtil
+import scalikejdbc.{SQLSyntaxSupport, SyntaxProvider, WrappedResultSet}
 
 /** Represents a set of identifiers as stored in DynamoDB */
```
Comment (kenoir) [inline common/src/main/scala/uk/ac/wellcome/models/Work.scala]: Not anymore :)

File: id_minter/src/test/scala/uk/ac/wellcome/platform/idminter/steps/IdentifierGeneratorTest.scala [hunk 2]
```diff
@@ -38,43 +40,31 @@ class IdentifierGeneratorTest
...
+        identifiers = List(SourceIdentifier("NotMiro", "NotMiroID", "1234")),
+        label = "some label")
     val futureId = identifierGenerator.generateId(work)
 
     whenReady(futureId.failed) { exception =>
       exception.getMessage shouldBe s"Item $work did not contain a MiroID"
     }
   }
-
-  it("should return an error if it finds more than one record for the same MiroID") {
```
Comment (kenoir) [inline id_minter/src/test/scala/uk/ac/wellcome/platform/idminter/steps/IdentifierGeneratorTest.scala]: Should we leave this one in?

Comment (alicefuzier) [inline id_minter/src/test/scala/uk/ac/wellcome/platform/idminter/steps/IdentifierGeneratorTest.scala]: It's impossible with Mysql, isn't it? We should add another one where we test that it throws an exception if we try to add a record for an already existing miroID

Comment (kenoir) [inline id_minter/src/test/scala/uk/ac/wellcome/platform/idminter/steps/IdentifierGeneratorTest.scala]: Ah, I thought was the intention of this test.

File: id_minter/src/main/scala/uk/ac/wellcome/platform/idminter/steps/IdentifierGenerator.scala [hunk 3]
```diff
@@ -1,82 +1,47 @@
...
 
-class IdentifierGenerator @Inject()(dynamoDBClient: AmazonDynamoDB,
-                                    dynamoConfig: DynamoConfig)
+class IdentifierGenerator @Inject()(identifiersDao: IdentifiersDao)
     extends Logging
     with TwitterModuleFlags {
 
-  private val identifiersTableName = dynamoConfig.table
-
   def generateId(work: Work): Future[String] =
```
Comment (alicefuzier) [inline id_minter/src/main/scala/uk/ac/wellcome/platform/idminter/steps/IdentifierGenerator.scala]: Test failed future if it cannot insert in the database

**Pull request merged**